snapcraft/utils.py

```diff
@@ -296,13 +297,75 @@ def get_ld_library_paths(prime_dir: Path, arch_triplet: str) -> str:
     """Return a usable in-snap LD_LIBRARY_PATH variable."""
     paths = ["${SNAP_LIBRARY_PATH}${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"]
     # Add the default LD_LIBRARY_PATH
-    paths += _get_common_ld_library_paths(prime_dir, arch_triplet)
+    paths += get_common_ld_library_paths(prime_dir, arch_triplet)
 
     ld_library_path = ":".join(paths)
 
     return re.sub(str(prime_dir), "$SNAP", ld_library_path)
 
 
+def get_host_tool(command_name: str) -> str:
+    """Return the full path of the given host tool.
+
+    :param command_name: the name of the command to resolve a path for.
+    :return: Path to command
+
+    :raises SnapcraftError: if command_name was not found.
+    """
+    tool = shutil.which(command_name)
+    if not tool:
+        raise errors.SnapcraftError(
+            f"A tool snapcraft depends on could not be found: {command_name!r}",
+            resolution="Ensure the tool is installed and available, and try again.",
+        )
+    return tool
+
+
+def get_snap_tool(command_name: str) -> str:
+    """Return the path command found in the snap.
+
+    If snapcraft is not running as a snap, shutil.which() is used
+    to resolve the command using PATH.
+
+    :param command_name: the name of the command to resolve a path for.
+    :return: Path to command
+
+    :raises SnapcraftError: if command_name was not found.
+    """
+    if os.environ.get("SNAP_NAME") != "snapcraft":
+        return get_host_tool(command_name)
+
+    snap_path = os.getenv("SNAP")
+    if snap_path is None:
+        raise RuntimeError("SNAP not defined, but SNAP_NAME is?")
```

Comment:
Pedantic: “SNAP [environmental variable] not defined,…”

I could see this error being confusing to someone who isn’t thinking about the SNAP environment variable.